### PR TITLE
Retain original provides metadata structure

### PR DESCRIPTION
**Pull request by ugexe**
Installing a distribution would change `provides => { a => "a.pm6" }`
to something like `provides => { a => { "a.pm6" => { ... } } }`,
with the purpose of saving some extra meta data about the file.

This removes the need for the additional meta data, and makes
installation no longer modify the provides structure. In the
future we may find the meta data useful again, but it should
be put into its own top level field.

### CODE REVIEW

File: src/core/CompUnit/Repository/Installation.pm6 [hunk 1]
```diff
@@ -11,13 +11,18 @@ class CompUnit::Repository::Installation does CompUnit::Repository::Locally does
...
+        nqp::sha1($name ~ $dist-id)
+    }
+
     submethod BUILD(:$!prefix, :$!lock, :$!WHICH, :$!next-repo --> Nil) { }
 
     my class InstalledDistribution is Distribution::Hash {
+        has $.dist-id;
         method content($address) {
-            my $entry = $.meta<provides>.values.first: { $_{$address}:exists };
+            my $entry = $.meta<provides>.antipairs.hash{$address};
```
Comment (niner) [inline src/core/CompUnit/Repository/Installation.pm6]: Does this actually work? We only upgrade repositories on write. So at this point we can still get the old meta data format.

Comment (ugexe) [reply src/core/CompUnit/Repository/Installation.pm6]: We upgrade repositories during `make install` too.

https://github.com/rakudo/rakudo/blob/68c2c9fea187fcdae33e841ae8a882d722e4804d/tools/build/Makefile-Moar.in#L330

**Comment by niner** [comment]
Upgrading during make install won't affect repos from a packaged rakudo and home or custom repos

**Comment by ugexe** [comment]
I would expect packaged rakudos to follow the upgrade repository procedure, even if they don't currently do so. existing `home` and `custom` repos are problematic. But at this point in time I expect the actual problems caused to people to be negligible, especially compared to the alternative (adding runtime checks whens searching to possible trigger repo upgrades).

**Comment by ugexe** [comment]
I suppose the various `Distribution` could have their constructor or `.meta` modified to return what "looks" like the original. But I'm not exactly a fan of this.

**Comment by niner** [comment]
On Saturday, 23 March 2019 18:59:29 CET Nick Logan wrote:
> I would expect packaged rakudos to follow the upgrade repository procedure,
> even if they don't currently do so.

On openSUSE it certainly does not. Actually in the case of the site repo, 
upgrading to a rakudo with version 3 repositories will break right now. Need 
to find a good solution for that. The vendor and perl repositories are not a 
problem, since the module packages will be rebuilt anyway.

> existing `home` and `custom` repos are
> problematic. But at this point in time I expect the actual problems caused
> to people to be negligible

I disagree. The code would after all return false data when run on a version 2 
repository, wouldn't it? I know that at work we have a couple of cases where 
modules are loaded out of the (service) user's home directory.

**Comment by ugexe** [comment]
I'm not sure that in reality this really breaks anything for existing repositories of modules since the json should not be getting read period expect in some very specific circumstances (that didn't really exist until the last release and the unification of Distribution handling between CURs).

**Comment by niner** [comment]
Ah ok, then it's much less of an issue indeed. Still need to figure out how to deal with repo version changes in a packaged rakudo though. Post install scripts are very much frowned upon and for good reason. I guess the rakudo package should not contain anything but the "site" directory itself but no contents at all.

**Comment by lizmat** [comment]
I feel this PR is in need of some love, or needs to be abandoned.

**Pull request closed without merging**